packages/react-components/react-tree/src/components/FlatTree/useHeadlessFlatTree.ts

```diff
@@ -181,7 +193,17 @@ export function useHeadlessFlatTree_unstable<Props extends HeadlessTreeItemProps
   const items = React.useCallback(() => headlessTree.visibleItems(openItems), [openItems, headlessTree]);
 
   return React.useMemo<HeadlessFlatTree<Props>>(
-    () => ({ navigate, getTreeProps, getNextNavigableItem, getElementFromItem, items }),
+    () => ({
```

Comment:
actually why does the return of `useHeadlessFlatTree` need to be memoized?

Reply:
technically, but for the virtualization scenario you might want to do something like this:

<img width="1193" alt="image" src="https://github.com/microsoft/fluentui/assets/5483269/d58171d0-ecfd-4823-b512-0ba6db716701">


And  `eslint/react-hooks` exhaustive deps rule erroneously complains on the dependency of a property in this case, saying that if you're using a property of an object you should memoize the whole object itself, so just to ensure that in this case we'll not be having any issues I memoized it also...